pkg/util/kube/security_context_test.go

```diff
@@ -30,40 +30,192 @@ func TestParseSecurityContext(t *testing.T) {
 		runAsUser                string
 		runAsGroup               string
 		allowPrivilegeEscalation string
+		secCtx                   string
 	}
 	tests := []struct {
 		name     string
 		args     args
 		wantErr  bool
 		expected *corev1.SecurityContext
 	}{
-		{"valid security context", args{"1001", "999", "true"}, false, &corev1.SecurityContext{
-			RunAsUser:                pointInt64(1001),
-			RunAsGroup:               pointInt64(999),
-			AllowPrivilegeEscalation: boolptr.True(),
-		}},
+		{
+			"valid security context",
+			args{"1001", "999", "true", ``},
+			false,
+			&corev1.SecurityContext{
+				RunAsUser:                pointInt64(1001),
+				RunAsGroup:               pointInt64(999),
+				AllowPrivilegeEscalation: boolptr.True(),
+			},
+		},
+		{
+			"valid security context with override runAsUser",
+			args{"1001", "999", "true", `runAsUser: 2000`},
+			false,
+			&corev1.SecurityContext{
+				RunAsUser:                pointInt64(2000),
+				RunAsGroup:               pointInt64(999),
+				AllowPrivilegeEscalation: boolptr.True(),
+			},
+		},
+		{
+			"valid securityContext with comments only secCtx key",
+			args{"", "", "", `
+capabilities:
+    drop:
+    - ALL
+    add:
+    - cap1
+    - cap2
+sELinuxOptions:
```

Comment:
Question, use "sELinuxOptions" is the same result as "seLinuxOptions"?
I saw the Kubernetes doc uses "seLinuxOptions".

Reply:
Absolutely that's a typo error ! I'll change it. Thanks for your attention.

Reply:
After some investigations, that was correct, yaml lib is not case sensitive ... so i you set `CaPAbiliTIes` as your yaml key it will fill the securityContext key `capabilities`.
I don't find a simple way to turn it case sensitive so i propose to keep that behavior (which is due to the standard json library) ?

Some details could be find here
https://pkg.go.dev/encoding/json#Unmarshal
```
Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match
```